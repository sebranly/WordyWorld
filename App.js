import React from "react";
import { StyleSheet, Text, View } from "react-native";

import SectionListBasics from "./SectionListBasics";

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <SectionListBasics letterCount={2} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  }
});
