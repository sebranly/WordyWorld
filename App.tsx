import * as React from "react";
import { StyleSheet, View } from "react-native";

import { SectionListBasics } from "./SectionListBasics";

export interface AppProps {}

const App: React.FC<AppProps> = _props => {
  return (
    <View style={styles.container}>
      <SectionListBasics letterCount={2} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  }
});

export default App;
