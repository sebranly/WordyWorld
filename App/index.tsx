// Vendor
import * as Font from "expo-font";
import * as React from "react";
import { AppLoading } from "expo";
import { Container } from "native-base";
import { Ionicons } from "@expo/vector-icons";
import { StyleSheet } from "react-native";

// Internal
import { About } from "../src/components/About";
import { Footer } from "../src/components/Footer";
import { Screen } from "../src/types/enum";
import { SectionListBasics } from "../src/components/SectionListBasics";
import { HEADER_Y, IS_TEST } from "../src/config/settings";

export interface AppProps {}

const App: React.FC<AppProps> = _props => {
  // Hooks
  const [isReady, setIsReady] = React.useState(false);
  const [screen, setScreen] = React.useState(Screen.Explore);

  // Life-cycle
  React.useEffect(() => {
    const loadFonts = async () => {
      await Font.loadAsync({
        Roboto: require("../node_modules/native-base/Fonts/Roboto.ttf"),
        Roboto_medium: require("../node_modules/native-base/Fonts/Roboto_medium.ttf"),
        ...Ionicons.font
      });

      setIsReady(true);
    };

    loadFonts();
  }, []); // Empty array leads to same behavior as `componentDidMount` (if it was a class)

  // Short-circuit
  if (!isReady && !IS_TEST) return <AppLoading />;

  // Handlers
  const changeScreen = (screen: Screen) => {
    setScreen(screen);
  };

  // Setup
  const isAbout = screen === Screen.About;
  const Component = isAbout ? <About /> : <SectionListBasics letterCount={3} />;

  // Markup
  return (
    <Container style={styles.container}>
      {Component}
      <Footer changeScreen={changeScreen} screen={screen} />
    </Container>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    flex: 1,
    paddingTop: HEADER_Y
  }
});

export default App;
