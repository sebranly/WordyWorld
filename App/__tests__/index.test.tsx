// Mocks
jest.mock("expo", () => ({ AppLoading: "AppLoading" }));
jest.mock("@expo/vector-icons", () => ({}));

// Vendor
import React from "react";
import { render } from "@testing-library/react-native";

// Internal
import App from "../index";

it("renders", () => {
  const props = { initialReady: true };
  const { container } = render(<App {...props} />);
  expect(container.children.length).toBeGreaterThan(0);
  expect(container.children).toMatchSnapshot();
});
