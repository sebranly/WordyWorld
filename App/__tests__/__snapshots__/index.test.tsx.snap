// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": 1334,
        },
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "paddingTop": 25,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "flex": 1,
          "height": 1334,
        }
      }
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Game
      </Text>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                s
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                p
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                l
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                a
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                s
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "grey",
                "flex": 1,
                "flexDirection": "column",
                "margin": 2,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 30,
                  }
                }
              >
                h
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "yellow",
            "flex": 1,
          }
        }
      >
        <Text>
          Example
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderColor": "#cbcbcb",
          "borderTopWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "left": 0,
          "paddingBottom": 0,
          "right": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": undefined,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <TouchableOpacity
          active={false}
          activeOpacity={0.5}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": "column",
              "height": null,
              "justifyContent": "center",
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          vertical={true}
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#6b6b6b",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Explore
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          active={true}
          activeOpacity={0.5}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#cde1f9",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": "column",
              "height": null,
              "justifyContent": "center",
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          vertical={true}
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#007aff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Play
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          active={false}
          activeOpacity={0.5}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": "column",
              "height": null,
              "justifyContent": "center",
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          vertical={true}
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#6b6b6b",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 16,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#6b6b6b",
                "fontFamily": "System",
                "fontSize": 14,
                "lineHeight": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            About
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  </View>,
]
`;
